tests/integration/common.go

```diff
@@ -0,0 +1,54 @@
+package integration
+
+import (
+	"errors"
+	"fmt"
+	"github.com/lf-edge/eden/pkg/adam"
+	"github.com/lf-edge/eden/pkg/utils"
+	uuid "github.com/satori/go.uuid"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+func adamPrepare() (adamCtx *adam.Ctx, id *uuid.UUID, err error) {
+	currentPath, err := os.Getwd()
+	if err != nil {
+		return nil, nil, err
+	}
+	ip := os.Getenv("IP")
+	if len(ip) == 0 {
+		ip, err = utils.GetIPForDockerAccess()
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+	port := os.Getenv("ADAM_PORT")
+	if len(port) == 0 {
+		port = "3333"
+	}
+	adamDir := os.Getenv("ADAM_DIST")
+	if len(adamDir) == 0 {
+		adamDir = path.Join(filepath.Dir(filepath.Dir(currentPath)), "dist", "adam")
+		if stat, err := os.Stat(adamDir); err != nil || !stat.IsDir() {
+			return nil, nil, err
+		}
+	}
+	ctx := adam.Ctx{
+		Dir: adamDir,
+		URL: fmt.Sprintf("https://%s:%s", ip, port),
+	}
+	cmdOut, err := ctx.DeviceList()
```

Comment:
I see that you implemented ctx.DeviceList by execing adam with a given options -- while this will work, I strongly suggest that you migrate to actually talking to Adam via HTTP channel:
    https://github.com/lf-edge/adam/blob/master/cmd/device.go#L41

Reply:
Done: https://github.com/lf-edge/eden/blob/90d45b9511149229f28d60d2d7dd6ffb45dfe068/pkg/controller/adam/adam.go#L62-L70 https://github.com/lf-edge/eden/blob/90d45b9511149229f28d60d2d7dd6ffb45dfe068/pkg/controller/adam/httpclient.go#L59-L77